src/ast/types.ts

```diff
@@ -295,6 +295,89 @@ export interface VariableDeclaration extends BaseDeclaration {
 
 export interface VariableDeclarator extends BaseNode {
   type: 'VariableDeclarator';
-  id: Identifier;
+  pattern: DeclaratorPattern;
   initialValue?: Expression;
 }
+
+export type DeclaratorPattern =
+  | ArrayPattern
+  | FunctionPattern
+  | Identifier
+  | PointerPattern;
+
+export interface PointerPattern extends BaseNode {
+  type: 'PointerPattern';
+  pattern: DeclaratorPattern;
+}
+
+export interface ArrayPattern extends BaseNode {
+  type: 'ArrayPattern';
+  id: DeclaratorPattern;
+  bracketContents: BracketContent[];
+}
+
+export type BracketContent =
+  | BracketExpressionContent
+  | BracketExpressionlessContent
+  | BracketStarContent;
+
+export interface BracketExpressionContent extends BaseNode {
+  type: 'BracketExpressionContent';
+  // TODO: Add type list when types are supported.
+  expression: Expression;
+  hasStaticBeforeTypes: boolean;
+  hasStaticAfterTypes: boolean;
```

Comment:
We can't have `static` both before and after, right? Would it be better to use a representation that captures this constraint?

Reply:
I changed this to be 1 field with an enum representing its status